provisioning/service/src/Contract/ContractApiHttp.cs

```diff
@@ -384,6 +383,126 @@ private static bool IsMappedToException(HttpResponseMessage message)
             return isMappedToException;
         }
 
+        /// <summary>
+        /// Unified HTTP request API
+        /// </summary>
+        /// <param name="httpMethod"></param>
+        /// <param name="requestUri"></param>
+        /// <param name="customHeaders"></param>
+        /// <param name="body"></param>
+        /// <param name="ifMatch"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        /// <exception cref="ProvisioningServiceClientException">if the cancellation was requested.</exception>
+        /// <exception cref="ProvisioningServiceClientTransportException">if there is a error in the HTTP communication 
+        ///     between client and service.</exception>
+        /// <exception cref="ProvisioningServiceClientTransientException">if the service answer the request with status 
+        ///     code 500 or higher. It means that the client must retry.</exception>
+        /// <exception cref="ProvisioningServiceClientBadUsageException">if the service answer the request with status 
+        ///     code between 400 and 499.</exception>
+        /// <exception cref="ProvisioningServiceClientServiceException">if the service answer the request with status 
+        ///     code between 300 and 399.</exception>
+        public async Task<ContractApiResponse> RequestAsync(
+            HttpMethod httpMethod,
+            Uri requestUri,
+            IDictionary<string, string> customHeaders,
+            string body,
+            string ifMatch,
+            CancellationToken cancellationToken)
+        {
+            ContractApiResponse response;
+
+            using (HttpRequestMessage msg = new HttpRequestMessage(httpMethod, requestUri))
+            {
+                msg.Content = new StringContent(body, Encoding.UTF8, CommonConstants.MediaTypeForDeviceManagementApis);
+
+                msg.Headers.Add(HttpRequestHeader.Authorization.ToString(), _authenticationHeaderProvider.GetAuthorizationHeader());
+                msg.Headers.Add(HttpRequestHeader.UserAgent.ToString(), Utils.GetClientVersion());
+
+                Task modifyRequestMessageAsync(HttpRequestMessage requestMsg, CancellationToken token)
+                {
+                    InsertIfMatch(requestMsg, ifMatch);
+                    AddCustomHeaders(requestMsg, customHeaders);
+                    return Task.FromResult(default(VoidTaskResult));
+                }
+
+                await modifyRequestMessageAsync(msg, cancellationToken);
+
+                try
+                {
+                    using (HttpResponseMessage httpResponse = await _httpClientObj.SendAsync(msg, cancellationToken))
```

Comment:
Whenever awaiting use .ConfigureAwait(false) in library code.

Reply:
Done

Reply:
FYI: my latest PR fixing the IoT Hub DeviceClient fixed ConfigureAwait everywhere else in the code.